### ISSUE TITLE
Default layer option does not work with multiple map instances

### ISSUE THREAD

**Issue by sgruenholz** [domoritz/leaflet-locatecontrol#146]
I have multiple map instances on a page, each using the locate control with default options.

When I click on the locate control of the first map, the first map pans correctly to the correct geolocation, but the marker and circle (this._layer) is added to the last map instead of the first one.

The problem is, this._layer is being set to a single layerGroup instance that is referenced by both maps instead of a new instance for each instantiation of the locate control.

Solution: change the default option from "new L.LayerGroup()" to "L.layerGroup()"
Hmm. This works for the first map, but now the last map does not work...

**Comment by domoritz** [comment]
You don't have to use the default. You can simply pass one layer to the control so that the same layer is used in all maps. 

I'm not sure I understand whether you want to sync the maps or not. The control is not really built to support multiple maps so you will need to listen for the locate events and sync the maps in your application code. 

let me know whether this answer makes sense and reopen if the shared layer does to work.

**Comment by sgruenholz** [comment]
I want each map to operate independently.

This is not happening currently. Instead, when I click on the locate control for map 1, it pans and zooms appropriately, but the marker and circle show up on map 2 instead. Somehow "this._layer" is getting shared between locate control instances so that the marker and circle only get added to the last map instantiated.

This happens regardless of what layer you use in options.

To fix it for my purposes I had to overwrite onAdd() to set:
this._layer = L.layerGroup 

instead of:
this._layer = this.options.layer

Anyhow, I have my solution, but I had to overwrite one of the core methods to make it happen.
I know that multiple map instances is not a common use case. Up to you if you want to deal with it or not. I'm guessing not.

I appreciate the timely response. Thanks for the awesome plugin!

Stephen
